Modules/_csv.c

```diff
@@ -1610,68 +1647,105 @@ static struct PyMethodDef csv_methods[] = {
     { NULL, NULL }
 };
 
-static struct PyModuleDef _csvmodule = {
-    PyModuleDef_HEAD_INIT,
-    "_csv",
-    csv_module_doc,
-    sizeof(_csvstate),
-    csv_methods,
-    NULL,
-    _csv_traverse,
-    _csv_clear,
-    _csv_free
-};
-
-PyMODINIT_FUNC
-PyInit__csv(void)
-{
-    PyObject *module;
+static int
+csv_exec(PyObject *module) {
     const StyleDesc *style;
+    PyObject *temp;
+    _csvstate *m_state = get_csv_state(module);
 
-    if (PyType_Ready(&Reader_Type) < 0)
-        return NULL;
+    temp = PyType_FromModuleAndSpec(module, &Dialect_Type_spec, NULL);
+    if (temp == NULL) {
+        return -1;
+    }
+    if (PyModule_AddObject(module, "Dialect", temp)) {
+        return -1;
+    }
+    Py_INCREF(temp);
+    m_state->dialect_type = (PyTypeObject *)temp;
 
-    if (PyType_Ready(&Writer_Type) < 0)
-        return NULL;
+    temp = PyType_FromModuleAndSpec(module, &Reader_Type_spec, NULL);
+    if (temp == NULL) {
+        return -1;
+    }
+    if (PyModule_AddObject(module, "Reader", temp)) {
```

Comment:
nitpick: Victor added a new perfect  `PyModule_AddObjectRef()` function in https://github.com/python/cpython/pull/23122
MAYBE you can try it in your PR :)

Reply:
I believe the tests wouldn't pass, because ` PyModule_AddObjectRef` is not part of the stable ABI on Windows (`PC/python3dll.c`).

Reply:
Sorry, that comment was for a different PR. Let me try this!